### ISSUE TITLE
Very high cpu usage

### ISSUE THREAD

**Issue by ksemaev** [algorand/go-algorand#1334]
### Subject of the issue
The cpu usage grew from 5 to ~20 cores.
As a side affect, I noticed that RAM usage also grew

### Your environment
* Software version: 
```
8589934600
2.0.8.stable [rel/stable] (commit #890f535b)
```
* Node status if applicable: 
```
Last committed block: 8238035
Time since last block: 0.9s
Sync Time: 0.0s
Last consensus protocol: https://github.com/algorandfoundation/specs/tree/e5f565421d720c6f75cdd186f7098495caf9101f
Next consensus protocol: https://github.com/algorandfoundation/specs/tree/e5f565421d720c6f75cdd186f7098495caf9101f
Round for next consensus protocol: 8238036
Next consensus protocol supported: true
Last Catchpoint: 8230000#HNUPHXXIFVQPG33S6BPJQICOMVDD7YVZDI463RQX2UFAN27CQEBQ
Genesis ID: mainnet-v1.0
Genesis hash: wGHE2Pwdvd7S12BL5FaOP20EGYesN73ktiC1qzkkit8=
```
* Operating System details:
``` debian:stretch docker container ```

### Steps to reproduce
Just run the archival node

### Expected behaviour
Average resource usage is about the same

### Actual behaviour
CPU and RAM consumption grows, disk usage dropped.

The config:
```
{
  "Version": 6,
  "AnnounceParticipationKey": true,
  "Archival": true,
  "BaseLoggerDebugLevel": 4,
  "BroadcastConnectionsLimit": -1,
  "CadaverSizeTarget": 1073741824,
  "CatchupFailurePeerRefreshRate": 10,
  "CatchupParallelBlocks": 16,
  "ConnectionsRateLimitingCount": 60,
  "ConnectionsRateLimitingWindowSeconds": 1,
  "DisableOutgoingConnectionThrottling": false,
  "DeadlockDetection": 0,
  "DNSBootstrapID": "<network>.algorand.network",
  "EnableIncomingMessageFilter": false,
  "EnableMetricReporting": false,
  "EnableOutgoingNetworkMessageFiltering": true,
  "EnableProfiler": false,
  "EnableRequestLogger": false,
  "EnableTopAccountsReporting": false,
  "EndpointAddress": "0.0.0.0:8080",
  "ForceRelayMessages": false,
  "GossipFanout": 4,
  "IncomingConnectionsLimit": 10000,
  "IncomingMessageFilterBucketCount": 5,
  "IncomingMessageFilterBucketSize": 512,
  "isIndexerActive": true,
  "LogArchiveMaxAge": "1h",
  "LogArchiveName": "node.archive.log",
  "LogSizeLimit": 10240,
  "MaxConnectionsPerIP": 30,
  "NetAddress": "",
  "NodeExporterListenAddress": ":9100",
  "NodeExporterPath": "./node_exporter",
  "OutgoingMessageFilterBucketCount": 3,
  "OutgoingMessageFilterBucketSize": 128,
  "PeerConnectionsUpdateInterval": 3600,
  "PriorityPeers": {},
  "ReconnectTime": 60000000000,
  "ReservedFDs": 256,
  "RestReadTimeoutSeconds": 15,
  "RestWriteTimeoutSeconds": 120,
  "RunHosted": false,
  "SuggestedFeeBlockHistory": 3,
  "SuggestedFeeSlidingWindowSize": 50,
  "TelemetryToLog": true,
  "TxPoolExponentialIncreaseFactor": 2,
  "TxPoolSize": 15000,
  "TxSyncIntervalSeconds": 60,
  "TxSyncServeResponseSize": 1000000,
  "TxSyncTimeoutSeconds": 30,
  "DNSSecurityFlags": 1,
  "EnablePingHandler": true
}
```

<img width="448" alt="Снимок экрана 2020-08-03 в 11 08 03" src="https://user-images.githubusercontent.com/20893598/89161153-e866d080-d57a-11ea-977d-f8ff74945542.png">
<img width="343" alt="Снимок экрана 2020-08-03 в 11 08 34" src="https://user-images.githubusercontent.com/20893598/89161159-e997fd80-d57a-11ea-9202-5ada22b64360.png">

Please tell:
1) What could cause that growing in resource usage?
2) Is there a way to limit resource usage (like num of cpu threads/cores etc)?

**Comment by Priegle** [comment]
@ksemaev - are you still seeing this? I believe we addressed in in 2.0.9 and would like to close the issue if possible.  Thanks!

**Comment by ksemaev** [comment]
@Priegle I switched to algo indexer&api, it's syncing now. I think the problem would be solved with that
